### ISSUE TITLE
How to access MariaDB(StatefulSets) from outside the cluster

### ISSUE THREAD

**Issue by udupicloud** [kubernetes/examples#439]
Hi Everyone,

**Things that I need to work with the StatefulSet pods**
I want to create a Production grade MariaDB Galera v10.6.5 cluster with Kubernetes (v1.19 -v1.23) and will use the Persistence Ceph Storage. Also I need that to be CentOS8 or CentOS Stream 8. Pods created via  StatefulSets should be accessible via SSH service from outside the cluster. 

**Things that I have tried**
1. I am able to create a StatefulSet with MariaDB Galera and Ceph as a PVC.
2. I have no issue with the deployment method single replica pods. 
3. I am able to ssh to the pods in Deployment and do necessary changes and able to run the commands of service like systemctl status MariaDB ,HTTPD.
4. I have tried to expose each pods with MetalLB load balancer. Also tried default NodePort and LoadBalancer. Able to access the pods with no deport but again how will the nginx or apache access it to store its data to MariaDB.
Ref : https://github.com/cgruver/mariadb-galera-okd
Ref : https://severalnines.com/database-blog/running-galera-cluster-kubernetes

**Note:- I need the StatefulSet pods to be accessible via SSH from outside the cluster.
 I should be able to use that pods for systemd commands to work for any applications that supports.** 

Basically I need help to create a yaml file for my requirements. Any help for the same would be of great help.

Thanking all in advance.
